bolt/lib/RuntimeLibs/RuntimeLibrary.cpp

```diff
@@ -38,13 +38,36 @@ std::string RuntimeLibrary::getLibPath(StringRef ToolPath,
     llvm::sys::path::append(LibPath, "lib" LLVM_LIBDIR_SUFFIX);
   }
   llvm::sys::path::append(LibPath, LibFileName);
-  if (!llvm::sys::fs::exists(LibPath)) {
-    errs() << "BOLT-ERROR: library not found: " << LibPath << "\n";
-    exit(1);
-  }
   return std::string(LibPath);
 }
 
+std::string RuntimeLibrary::getLibPathByInstalled(StringRef LibFileName) {
+  SmallString<128> LibPath(CMAKE_INSTALL_FULL_LIBDIR);
```

Comment:
This embeds a full absolute path into the bolt binary. That doesn't seem great, since it means that that library has to be at that particular path. Other tools instead require a fixed relative path from executable to runtime library (with an overridable setting).

See `Driver::GetResourcesPath` in clang/lib/Driver/Driver.cpp (and its caller, in the same file), and `ToolChain::getCompilerRTPath()` in clang/lib/Driver/ToolChain.cpp.

Reply:
This particular method is meant only for when the binary is installed. `getLibPathByToolPath` gets the path relative to the binary. `getLibPath` calls both methods to ensure it finds the right library.

Reply:
"Installed" isn't a concept that always makes sense. On Windows, there's no system install. Even on other platforms, projects might download a hermetic toolchain to a random subdirectory.